docker/configure_workers_and_start.py

```diff
@@ -50,7 +50,13 @@
 
 MAIN_PROCESS_HTTP_LISTENER_PORT = 8080
 
-
+# Workers with exposed endpoints needs either "client", "federation", or "media" listener_resources
+# Watching /_matrix/client needs a "client" listener
+# Watching /_matrix/federation needs a "federation" listener
+# Watching /_matrix/media and related needs a "media" listener
+# Stream Writers require "client" and "replication" listeners because they
+#   have to attach by instance_map to the master process and have client endpoints.
+# BIG WARNING: typing and receipts stream writers are not working correctly at this time.
 WORKERS_CONFIG: Dict[str, Dict[str, Any]] = {
```

Comment:
I am confused by the warning? What exactly does this mean?

Reply:
It's slightly out of date. Typing appears to be fine on my live server but receipts crashes the system. It's a separate issue that I'm not completely sure how to write down yet. I'm still learning how to do somethings. So, big warning in advance that "this in particular doesn't work right yet". It can be removed if you think it's inappropriate, or after it's fixed.

Reply:
Ah okay. In that case, if adding the typing and receipts stream readers doesn't currently work, I'd either figure out how to fix them or remove them from the PR. In general you never want to merge code you are not positive that it works. 
> I'm still learning how to do somethings. 
No worries! Just taking a stab at things is the best way to learn :)

Reply:
You should see some of the "pseudo-fuzzying" I've done to figure out how things work. I've read _A LOT_ while following linux kernel development in the past. It's all done in my personal fork of synapse, but it's not thoroughly documented.

Reply:
Perhaps for the purpose of not having to do it again in the future, I could hide it in a commented out section? I've seen that done somewhere in other's pull requests. If the history is preserved here, can always look it up again if not.

Reply:
Also this:
>Final testing results on live: Everything appears to function(or at least not be a show stopper) except for the receipts worker type which did bring down the whole synapse server. This may be a red herring, as Complement tests still pass and may just be an issue with migrating a live server with existing database. Can anyone provide me with a way to check this?